tests/test_socket_server.py

```diff
@@ -115,6 +118,11 @@ async def test_exit(self) -> None:
         self.connected_future = asyncio.Future()
         assert self.srv.connected
         await self.assert_configure(name="TEST_EXIT")
+        # Read at least one line to make sure that the MockDevice background
+        # thread can be stopped.
+        await self.read()
+        # Make sure that all background threads of mock devices are stopped.
+        await self.write(command=common.Command.DISCONNECT, parameters={})
```

Comment:
Does this mean we can't just disconnect from the RPi data server? (I suppose the fact that you were already calling it at line 112 suggests the same thing).

Reply:
Thanks for catching that. I added those 5 lines (and three similar ones in another test method in this class) when I was playing with multi-threading and forgot to remove them when I reverted all that. So I removed all of them now.